src/session.ts

```diff
@@ -0,0 +1,278 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+import utils = require('./utils')
+import kernel = require('./kernel')
+
+import Signal = phosphor.core.Signal;
+import emit = phosphor.core.emit;
+import IAjaxSuccess = utils.IAjaxSuccess;
+import IAjaxError = utils.IAjaxError;
+
+
+/**
+ * Notebook Identification specification.
+ */
+export
+interface INotebookId {
+  path: string;
+};
+
+
+/**
+ * Session Identification specification.
+ */
+export
+interface ISessionId {
+  id: string;
+  notebook: INotebookId;
+  kernel: kernel.IKernelId;
+};
+
+
+/**
+ * Session initialization options.
+ */
+export
+interface ISessionOptions {
+  notebookPath: string;
+  kernelName: string;
+  baseUrl: string;
+  wsUrl: string;
+};
+
+
+/**
+ * Session object for accessing the session REST api. The session
+ * should be used to start kernels and then shut them down -- for
+ * all other operations, the kernel object should be used.
+ **/
+export 
+class Session {
+
+  static statusChanged = new Signal<Session, string>();
+
+  /**
+   * GET /api/sessions
+   *
+   * Get a list of the current sessions.
+   */
+  static list(sessionServiceUrl: string): Promise<ISessionId[]> {
+    return utils.ajaxRequest(sessionServiceUrl, {
+      method: "GET",
+      dataType: "json"
+    }).then((success: IAjaxSuccess): ISessionId[] => {
+      if (success.xhr.status === 200) {
```

Comment:
We could add another argument to `AjaxRequest` with a list of valid status codes.  It would break from the jQuery syntax, but I don't necessarily see that as a bad thing.